File: 프로그래머스 풀이/lv1_로또의_최고_순위와_최저_순위.py
def solution(lottos, win_nums):
    lottos.sort()
    win_nums.sort()

    cnt_zero = lottos.count(0)
    sum = 0    
    for win in win_nums :     # lotto 번호는 6가지이다.
        try :
            start = lottos.index(win)+1
            sum += 1
        except : 
            start = 0
        lottos = lottos[start:]     # sort되어있기 때문에 앞에 글자는 없어도 된다.
    
    answer = [sum+cnt_zero, sum]
    return [6 if num < 2 else 7-num for num in answer ]

    
        
    return answer

File: 프로그래머스 풀이/lv1_로또의_최고_순위와_최저_순위.java
import java.util.Arrays;
class Solution {
    public int[] ret_grade(int grade, int zero) {
        if ( grade == 7 ) {
            if ( zero == 0 )    return new int[]{6, 6};
            return new int[]{grade-zero, 6};
        }
        if ( grade == 1 )    return new int[]{grade, grade};
        return new int[]{grade-zero, grade};
    
    }
    public int[] solution(int[] lottos, int[] win_nums) {
        int correct = 0;
        Arrays.sort(lottos);
        Arrays.sort(win_nums);
        
        int idx = 0, zero_num = 0, w_idx=0, comp;
        while (idx < 6 && lottos[idx] == 0 ){
            zero_num++; idx++;
        }
        while ( idx < 6 && w_idx < 6 ) {
            comp = Integer.compare(lottos[idx], win_nums[w_idx]);
            if ( comp == 0 ){
                correct++; idx++; w_idx++;
            }
            else if ( comp == -1 )  idx++;
            else    w_idx++;
        }

        return ret_grade(7 - correct, zero_num);
    }
}